pkg/test/framework/components/namespace/namespace.go

```diff
@@ -22,6 +22,29 @@ import (
 	"istio.io/istio/pkg/test/framework/resource"
 )
 
+// Config contains configuration information about the namespace instance
+// including naming information, sidecar injection config, and other component-specific labels
+// note: each component that adds additional custom k8s labels for control should have a field here
+type Config struct {
+	Prefix        string
+	SidecarConfig SidecarConfig
+	CitadelConfig CitadelConfig
+}
+
+// CitadelConfig contains the values for configuration of citadel-specific namespace labels
+// if not included in config, namespace will have none of these labels
+type CitadelConfig struct {
+	ManagedLabel  string
+	OverrideLabel string
+}
+
+// SidecarConfig contains the values for configuration of sidecar-specific namespace labels
+// if not included in config, namespace will have none of these labels
+type SidecarConfig struct {
```

Comment:
I'd be fine with `InjectionConfig` if you really want a sub-struct, but I would argue that it's uglier from the call sites.  I think it would be better to just define everything at the top-level.

Reply:
I see your point, I'm fine with making these fields top-level.